File: frontend/src/features/teams/types/team.ts
export type TeamRole = 'admin' | 'member';

export interface TeamMember {
  id: number;
  team_id: number;
  user_id: number;
  user_name: string;
  email: string;
  role: TeamRole;
  joined_at: string;
}

export interface Team {
  id: number;
  organization_id: number;
  name: string;
  lead_id: number;
  lead_name?: string;
  is_active: boolean;
  member_count?: number;
  created_at: string;
  updated_at: string;
  members?: TeamMember[];
}

// NEW — what /teams/:id returns (team + guaranteed members)
export interface TeamDetail extends Team {
  members: TeamMember[];
}

export interface CreateTeamRequest {
  organization_id: number;
  name: string;
  lead_id?: number;
}

export interface UpdateTeamRequest {
  name?: string;
  lead_id?: number;
}

export interface AddTeamMemberRequest {
  user_id: number;
  role?: TeamRole;
}

export interface UpdateTeamMemberRoleRequest {
  role: TeamRole;
}

File: backend/internal/model/team.go
package model

import (
	"time"
)

type Team struct {
	ID             int        `gorm:"primaryKey;autoIncrement"`
	OrganizationID int        `gorm:"not null;index"`
	Name           string     `gorm:"size:100;not null"`
	LeadID         int        `gorm:"not null;index"`
	IsActive       bool       `gorm:"default:true"`
	CreatedAt      time.Time  `gorm:"autoCreateTime"`
	UpdatedAt      time.Time  `gorm:"autoUpdateTime"`
 
	Organization Organization    `gorm:"foreignKey:OrganizationID"`
	Lead         User            `gorm:"foreignKey:LeadID"`
	Members      []TeamMember    `gorm:"foreignKey:TeamID"`
}

func (Team) TableName() string {
	return "teams"
}

type TeamMember struct {
	ID        int       `gorm:"primaryKey;autoIncrement"`
	TeamID    int       `gorm:"not null;index"`
	UserID    int       `gorm:"not null;index"`
	Role      string    `gorm:"size:20;default:'member'"`
	JoinedAt  time.Time `gorm:"autoCreateTime"`
	IsActive  bool      `gorm:"default:true"`
	CreatedAt time.Time `gorm:"autoCreateTime"`
	UpdatedAt time.Time `gorm:"autoUpdateTime"`
 
	Team Team `gorm:"foreignKey:TeamID"`
	User User `gorm:"foreignKey:UserID"`
}

func (TeamMember) TableName() string {
	return "team_members"
}
 
const (
	TeamRoleAdmin  = "admin"
	TeamRoleMember = "member"
)